main.js

```diff
@@ -41,56 +47,113 @@ const drawBall = () => {
 };
 
 const updatePaddles = () => {
-  console.log(paddleOneSpeed);
-
-  paddleOneSpeed -= paddleOneSpeed * friction;
-
-  if (paddleOneSpeed > maxSpeed) {
-    paddleOneSpeed = maxSpeed;
-  }
-  if (paddleOneSpeed < -maxSpeed) {
-    paddleOneSpeed = -maxSpeed;
-  }
-  if (paddleTwoSpeed > maxSpeed) {
-    paddleTwoSpeed = maxSpeed;
-  }
-  if (paddleTwoSpeed < -maxSpeed) {
-    paddleTwoSpeed = -maxSpeed;
-  }
+  paddleOneSpeed -= friction * paddleOneSpeed;
+  paddleTwoSpeed -= friction * paddleTwoSpeed;
 
+  //if they are moving pretty slow, just stop them.
   if (Math.abs(paddleOneSpeed) < friction) {
     paddleOneSpeed = 0;
   }
-
   if (Math.abs(paddleTwoSpeed) < friction) {
     paddleTwoSpeed = 0;
   }
 
+  //add intertia from keypresses
+  if (paddleOneControls.up) {
+    paddleOneSpeed -= acceleration;
+  }
+  if (paddleOneControls.down) {
+    paddleOneSpeed += acceleration;
+  }
+  if (paddleTwoControls.up) {
+    paddleTwoSpeed -= acceleration;
+  }
+  if (paddleTwoControls.down) {
+    paddleTwoSpeed += acceleration;
+  }
+
   paddleOne.y += paddleOneSpeed;
   paddleTwo.y += paddleTwoSpeed;
 
+  //bounce off the top and bottom
   if (paddleOne.y < 0) {
     paddleOne.y = 0;
+    paddleOneSpeed *= -1;
   }
   if (paddleOne.y > canvas.height - paddleOne.height) {
     paddleOne.y = canvas.height - paddleOne.height;
+    paddleOneSpeed *= -1;
   }
   if (paddleTwo.y < 0) {
     paddleTwo.y = 0;
+    paddleTwoSpeed *= -1;
   }
   if (paddleTwo.y > canvas.height - paddleTwo.height) {
     paddleTwo.y = canvas.height - paddleTwo.height;
+    paddleTwoSpeed *= -1;
   }
 };
 
 const updateBall = () => {
+  if (justHit) {
+    if (ball.x < buffer) {
+      ballSpeedHorizontal = Math.abs(ballSpeedHorizontal);
+    } else if (ball.x + buffer > canvas.width) {
+      ballSpeedHorizontal = -1 * Math.abs(ballSpeedHorizontal);
+    } else {
+      justHit = false;
+    }
+  }
   ball.x += ballSpeedHorizontal;
   ball.y += ballSpeedVertical;
-  if (ball.x < ball.radius || ball.x > canvas.width - ball.radius) {
-    ballSpeedHorizontal *= -1;
+
+  //scoring
+  if (ball.x - border < ball.radius) {
+    ballSpeedHorizontal = Math.abs(ballSpeedHorizontal + Math.random() / 50);
+    paddleTwo.height -= evaporation;
+  }
+  if (ball.x + border > canvas.width - ball.radius) {
+    ballSpeedHorizontal = -1 * Math.abs(ballSpeedHorizontal + Math.random() / 50);
+    paddleOne.height -= evaporation;
+  }
+
+  //is the game over?
+  if (paddleOne.height <= 0) {
+    alert('Player Left is the Winner!  Click okay to replay!');
+    location.reload();
```

Comment:
I was thinking more like making a function that would reset player heights to the starting values + put ball in the middle. I guess this works though